acvm-repo/acvm/tests/solver.rs

```diff
@@ -841,9 +1104,161 @@ prop_compose! {
+
+    let internal_expected_result = vec![
+        field_from_hex("18DFB8DC9B82229CFF974EFEFC8DF78B1CE96D9D844236B496785C698BC6732E"),
+        field_from_hex("095C230D1D37A246E8D2D5A63B165FE0FADE040D442F61E25F0590E5FB76F839"),
+        field_from_hex("0BB9545846E1AFA4FA3C97414A60A20FC4949F537A68CCECA34C5CE71E28AA59"),
+        field_from_hex("18A4F34C9C6F99335FF7638B82AEED9018026618358873C982BBDDE265B2ED6D"),
+    ];
+
+    assert_eq!(expected_result, into_repr_vec(internal_expected_result));
+    assert_eq!(result, expected_result);
+}
+
+#[test]
+fn sha256_zeros() {
+    let results = solve_array_input_blackbox_call(vec![], 32, sha256_op);
+    let expected_results: Vec<_> = vec![
+        227, 176, 196, 66, 152, 252, 28, 20, 154, 251, 244, 200, 153, 111, 185, 36, 39, 174, 65,
+        228, 100, 155, 147, 76, 164, 149, 153, 27, 120, 82, 184, 85,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+#[test]
+fn sha256_compression_zeros() {
+    let results = solve_array_input_blackbox_call(
+        [(FieldElement::zero(), false); 24].try_into().unwrap(),
+        8,
+        sha256_compression_op,
+    );
+    let expected_results: Vec<_> = vec![
+        2091193876, 1113340840, 3461668143, 3254913767, 3068490961, 2551409935, 2927503052,
+        3205228454,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+#[test]
+fn blake2s_zeros() {
+    let results = solve_array_input_blackbox_call(vec![], 32, blake2s_op);
+    let expected_results: Vec<_> = vec![
+        105, 33, 122, 48, 121, 144, 128, 148, 225, 17, 33, 208, 66, 53, 74, 124, 31, 85, 182, 72,
+        44, 161, 165, 30, 27, 37, 13, 253, 30, 208, 238, 249,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+#[test]
+fn blake3_zeros() {
+    let results = solve_array_input_blackbox_call(vec![], 32, blake3_op);
+    let expected_results: Vec<_> = vec![
+        175, 19, 73, 185, 245, 249, 161, 166, 160, 64, 77, 234, 54, 220, 201, 73, 155, 203, 37,
+        201, 173, 193, 18, 183, 204, 154, 147, 202, 228, 31, 50, 98,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+#[test]
+fn keccak256_zeros() {
+    let results = solve_array_input_blackbox_call(vec![], 32, keccak256_op);
+    let expected_results: Vec<_> = vec![
+        197, 210, 70, 1, 134, 247, 35, 60, 146, 126, 125, 178, 220, 199, 3, 192, 229, 0, 182, 83,
+        202, 130, 39, 59, 123, 250, 216, 4, 93, 133, 164, 112,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+// TODO(https://github.com/noir-lang/noir/issues/5687): internal error when calling Keccakf1600
```

Comment:
`num_bits` on the failing assert in the issue comes from a `FunctionInput` type. I think this is being triggered by this line:
```
FunctionInput::witness(Witness((i + offset) as u32), FieldElement::max_num_bits())
```
in `constant_or_witness_to_function_inputs` always setting a function input bit size. I'm not sure if this is an internal error due to this reason. We should change the set-up in these tests to account for the function input bit size.

Reply:
We can perhaps change the ACVM to return an error that codegen was incorrect, but this panic looks to be hit due to incorrect set-up not due to the solver.

Reply:
Ok, will do :+1:

This will make it easier to tell which blackbox functions are affected: it seems like most of them work with `FieldElement::max_num_bits()`.

Reply:
This one's now passing 🙏